packages/storefront-lwc/src/modules/commerce/basket/__tests__/basket.test.js

```diff
@@ -0,0 +1,276 @@
+
+function tick(time = null) {
+    return new Promise(resolve => setTimeout(resolve, time));
+}
+
+describe('<commerce-basket>', function() {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+
+        reset();
+    });
+
+    it('should render basket loading', async function() {
+        const element = createElement('commerce-basket', {
+            is: Basket,
+        });
+
+        document.body.appendChild(element);
+
+        await tick();
+
+        expect(element.shadowRoot.querySelector('.loading')).toMatchSnapshot();
+    });
+
+    it('should render an empty basket', async function() {
+        mockQuery(emptyBasketResponse);
+
+        const element = createElement('commerce-basket', {
+            is: Basket,
+        });
+
+        document.body.appendChild(element);
+
+        await tick();
+
+        expect(
+            element.shadowRoot.querySelector('.text-empty-basket'),
+        ).toMatchSnapshot();
+    });
+
+    it('should render a basket with an item', async function() {
+        mockQuery(basketWithItem);
+
+        const element = createElement('commerce-basket', {
+            is: Basket,
+        });
+
+        document.body.appendChild(element);
+
+        await tick();
+
+        expect(
+            element.shadowRoot.querySelector('.header-title .number-of-items'),
+        ).toMatchSnapshot();
+    });
+
+    it('should update basket details', async function() {
+        mockQuery(basketWithItem);
+
+        const element = createElement('commerce-basket', {
+            is: Basket,
+        });
+
+        document.body.appendChild(element);
+
+        await tick();
+
+        element.shadowRoot
+            .querySelector('commerce-shippingmethods')
+            .dispatchEvent(
+                new CustomEvent('updateshippingmethod', {
+                    detail: {
+                        updatedBasket: {
+                            ...basketWithItem.data.getBasket,
+                            totalProductsQuantity: 2,
+                        },
+                    },
+                }),
+            );
+
+        await tick();
+
+        expect(
+            element.shadowRoot.querySelector('.header-title .number-of-items'),
+        ).toMatchSnapshot();
+    });
+});
+
+const emptyBasketResponse = {
+    errors: [
+        {
+            message: 'No Basket has been created yet.',
+            locations: [{ line: 2, column: 3 }],
+            path: ['getBasket'],
+            extensions: {
+                code: 'INTERNAL_SERVER_ERROR',
+                exception: {
+                    stacktrace: [
+                        'Error: No Basket has been created yet.',
+                        '    at getBasket (/home/blittle/dev/sfcc-sample-apps/packages/@sfcc-bff/basketapi/dist/basketapi.cjs.development.js:375:13)',
```

Comment:
Do we need matching the stack trace? It will not pass on another environment like this.

Reply:
It doesn't really matter, it is just a stack. I removed it out though.